lib/instrumentation/mysql/mysql.js

```diff
@@ -11,63 +11,116 @@ const symbols = require('../../symbols')
 const { QuerySpec } = require('../../shim/specs')
 const DatastoreParameters = require('../../shim/specs/params/datastore')
 
+/**
+ * Used to instrument `mysql` and `mysql2` packages
+ * the `mysql2/promise` instrumentation is below in `promiseInitialize`
+ *
+ * @param {Shim} shim instance of shim
+ * @param {object} mysql package to instrument
+ */
 function callbackInitialize(shim, mysql) {
   shim.setDatastore(shim.MYSQL)
   shim[symbols.wrappedPoolConnection] = false
 
   shim.wrapReturn(mysql, 'createConnection', wrapCreateConnection)
-  function wrapCreateConnection(shim, fn, fnName, connection) {
-    if (shim[symbols.unwrapConnection]) {
-      return
-    }
-    shim.logger.debug('Wrapping Connection#query')
-    if (wrapQueryable(shim, connection, false)) {
-      const connProto = Object.getPrototypeOf(connection)
-      connProto[symbols.storeDatabase] = true
-      shim[symbols.unwrapConnection] = true
+  shim.wrapReturn(mysql, 'createPool', wrapCreatePool)
+  shim.wrapReturn(mysql, 'createPoolCluster', wrapCreatePoolCluster)
+}
+
+/**
+ * Used to instrument `mysql2/promise`
+ *
+ * @param {Shim} shim instance of shim
+ * @param {object} mysql2 package to instrument
+ */
+function promiseInitialize(shim, mysql2) {
+  shim.setDatastore(shim.MYSQL)
+  shim[symbols.wrappedPoolConnection] = false
+  shim.wrap(
```

Comment:
Should this, and the others in this change set, be `shim.wrapReturn`?

Reply:
no because in this case we have the function we need to wrap directly. In the callback style it's what gets returned from the `createConnection`, `createPool`, etc